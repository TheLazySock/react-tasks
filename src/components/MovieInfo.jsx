import React from 'react';
import { Link } from 'react-router-dom';

import Loader from '../containers/Loader.jsx';

const MovieInfo = (props) => {
    const newLocal = props.movie; // линтер ругался на movie = props.movie, исправил как-то так.
    const movie = newLocal;
    const releaseDateYear = new Date(movie.release_date).getFullYear();

    const genresString = Array.isArray(movie.genres) ? movie.genres.join(' & ') : 'No any genres';

    return (
        <div className="backdrop">
            <header className="backdrop-header">
                <Link className="site-name" to={'/search'}>netflixroulette</Link>
                <Link className="search-link" to={'/search'}>SEARCH</Link>
            </header>
            <div className="movie">
                <Loader loading={props.loading}>
                    <img src={movie.poster_path} className="movie-image" />
                    <div className="movie-details">
                        <div className="movie-title">{movie.title}</div>
                        <div className="movie-rating">{movie.vote_average}</div>
                        <div className="movie-genre">{genresString}</div>
                        <div className="movie-numbers">{releaseDateYear}</div>
                        <div className="movie-numbers">{movie.runtime} min</div>
                        <div className="scrollable">
                            <div className="movie-description">{movie.overview}</div>
                        </div>
                    </div>
                </Loader>
            </div>
        </div>
    );
};

export default MovieInfo;
