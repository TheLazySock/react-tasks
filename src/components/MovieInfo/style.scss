@import "../../containers/SearchPage/style.scss";

.movie {
    display: flex;
    flex-direction: row;
    color: #fefefe;
    margin: auto;
    max-width: 80%;
    .loader {
        display: inherit;
        flex-direction: row;
        margin: auto;
        width: 100%;
        min-height: 430px;
    }
    .movie-image {
        width: 220px;
        height: 330px;
        flex: 1 1 auto;
        object-fit: contain;
        margin: 40px;
    }
    .movie-details {
        flex: 1 1 auto;
        margin: 40px 80px 40px 0;
        .movie-title {
            display: inline-block;
            color: #f55263;
            font-size: 24px;
            line-height: 40px;
            vertical-align: middle;
            font-weight: bold;
        }
        .movie-rating {
            color: #90ee90;
            display: inline-block;
            border: solid 1px #90ee90;
            border-radius: 40px;
            width: 40px;
            line-height: 40px;
            font-weight: bold;
            text-align: center;
            margin-left: 20px;
        }
        .movie-numbers {
            display: inline-block;
            padding: 10px 20px 10px 0px;
            font-size: 12px;
            font-weight: bold;
        }
        .movie-description {
            margin: 10px 0;
            // max-width: 70%;
        }
        .movie-role {
            font-size: 11px;
            margin-top: 30px;
        }
        .scrollable{
            max-height: 200px;
        }
    }
}