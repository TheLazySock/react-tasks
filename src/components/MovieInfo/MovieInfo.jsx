// @flow
import React from 'react';
import { Link } from 'react-router-dom';

import Loader from '../../containers/Loader';
import style from './style.scss';

type MovieInfoProps = {
    loading: boolean,
    movie: Object
}

const MovieInfo = (props: MovieInfoProps) => {
    const newLocal: Object = props.movie; // линтер ругался на movie = props.movie, исправил как-то так.
    const movie: {
        release_date: Date,
        genres: Array<string>,
        poster_path: string,
        title: string,
        overview: string,
        vote_average: number,
        runtime: number
    } = newLocal;
    const releaseDateYear: number | string = new Date(movie.release_date).getFullYear();

    const genresString: string = Array.isArray(movie.genres) ? movie.genres.join(' & ') : 'No any genres';

    return (
        <div className={style.backdrop}>
            <header className={style['backdrop-header']}>
                <Link className={style['site-name']} to={'/search'}>netflixroulette</Link>
                <Link className={style['search-link']} to={'/search'}>SEARCH</Link>
            </header>
            <div className={style.movie}>
                <Loader loading={props.loading}>
                    <img src={movie.poster_path} className={style['movie-image']} />
                    <div className={style['movie-details']}>
                        <div className={style['movie-title']}>{movie.title}</div>
                        <div className={style['movie-rating']}>{movie.vote_average}</div>
                        <div className={style['movie-genre']}>{genresString}</div>
                        <div className={style['movie-numbers']}>{releaseDateYear}</div>
                        <div className={style['movie-numbers']}>{movie.runtime} min</div>
                        <div className={style.scrollable}>
                            <div className={style['movie-description']}>{movie.overview}</div>
                        </div>
                    </div>
                </Loader>
            </div>
        </div>
    );
};

export default MovieInfo;
