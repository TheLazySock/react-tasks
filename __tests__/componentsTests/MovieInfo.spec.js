import React from 'react';
import renderer from 'react-test-renderer';
import MovieInfo from '../../src/components/MovieInfo';
import { shallow, configure } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('>>>MOVIE INFO --- snapshot', () => {
    const movieInfoProps = {
        poster_path: "https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg",
        title: "Pulp Fiction",
        release_date: "1994-09-10",
        genres: ["Thriller", "Crime"],
        vote_average: 8.3,
        runtime: 154,
        overview: "A burger-loving hit man, his philosophical partner, " 
                + "a drug-addled gangster's moll and a washed-up boxer converge " 
                + "in this sprawling, comedic crime caper. Their adventures " 
                + "unfurl in three stories that ingeniously trip back and forth in time."
    }
    let container;

    beforeEach(() => {
        container = shallow(<MovieInfo movie={movieInfoProps}/>);
    });

    it('MovieInfo have to rendered succesfully', () => {
        expect(shallowToJson(container)).toMatchSnapshot();
    });

    it('MovieInfo genres should work correctly', () => {
        expect(container.find('.movie-genre').text()).toBe('Thriller & Crime');
    });

    it('Loader should work', () => {
        const container = shallow(<MovieInfo movie={movieInfoProps} loading={true} />);
        expect(container.find('Loader')).toHaveLength(1);
    })
});