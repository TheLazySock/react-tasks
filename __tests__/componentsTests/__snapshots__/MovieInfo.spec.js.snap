// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>>>MOVIE INFO --- snapshot MovieInfo have to rendered succesfully 1`] = `
<div
  className="backdrop"
>
  <header
    className="backdrop-header"
  >
    <Link
      className="site-name"
      replace={false}
      to="/search"
    >
      netflixroulette
    </Link>
    <Link
      className="search-link"
      replace={false}
      to="/search"
    >
      SEARCH
    </Link>
  </header>
  <div
    className="movie"
  >
    <Loader>
      <img
        className="movie-image"
        src="https://image.tmdb.org/t/p/w500/dM2w364MScsjFf8pfMbaWUcWrR.jpg"
      />
      <div
        className="movie-details"
      >
        <div
          className="movie-title"
        >
          Pulp Fiction
        </div>
        <div
          className="movie-rating"
        >
          8.3
        </div>
        <div
          className="movie-genre"
        >
          Thriller & Crime
        </div>
        <div
          className="movie-numbers"
        >
          1994
        </div>
        <div
          className="movie-numbers"
        >
          154
           min
        </div>
        <div
          className="scrollable"
        >
          <div
            className="movie-description"
          >
            A burger-loving hit man, his philosophical partner, a drug-addled gangster's moll and a washed-up boxer converge in this sprawling, comedic crime caper. Their adventures unfurl in three stories that ingeniously trip back and forth in time.
          </div>
        </div>
      </div>
    </Loader>
  </div>
</div>
`;
